packages/forms/test/form_group_spec.ts

```diff
@@ -1825,6 +1825,386 @@ describe('FormGroup', () => {
+           ]);
+
+           attachEventsLogger(g1, logs, 'g1');
+
+           const c2 =
+               new FormControl('new!', null, simpleAsyncValidator({timeout: 1, shouldFail: true}));
+
+           attachEventsLogger(c2, logs, 'c2');
+
+           // Initial state of the new control
+           expect(currentStateOf([c2])).toEqual([
+             {errors: null, pending: true, status: 'PENDING'},  // Control 2
+           ]);
+
+           expect(logs.length).toBe(0);
+
+           g1.setControl('one', c2);
+
+           tick(1);
+
+           expect(logs).toEqual([
+             'value (g1): {"one":"new!"}',  // value change emitted by `setControl`
+             'status (g1): PENDING',        // value change emitted by `setControl`
+             'status (c2): INVALID',        // async validator run after `setControl` call
+             'status (g1): INVALID'         // async validator run after `setControl` call
+           ]);
+
+           // Final state of all controls
+           expect(currentStateOf([c1, g1, c2])).toEqual([
+             {errors: null, pending: false, status: 'VALID'},             // Control 1
+             {errors: null, pending: false, status: 'INVALID'},           // Group
+             {errors: {async: true}, pending: false, status: 'INVALID'},  // Control 2
+           ]);
+         }));
+
+      it('should run the async validator at `FormControl` and `FormGroup` level and set status to `INVALID`',
+         fakeAsync(() => {
+           const logs: string[] = [];
+           const c1 = new FormControl('one');
+           const g1 = new FormGroup(
+               {'one': c1}, null, simpleAsyncValidator({timeout: 1, shouldFail: true}));
+
+           // Initial state of the controls
+           expect(currentStateOf([c1, g1])).toEqual([
+             {errors: null, pending: false, status: 'VALID'},   // Control 1
+             {errors: null, pending: true, status: 'PENDING'},  // Group
+           ]);
+
+           attachEventsLogger(g1, logs, 'g1');
+
+           const c2 =
+               new FormControl('new!', null, simpleAsyncValidator({timeout: 1, shouldFail: true}));
+
+           attachEventsLogger(c2, logs, 'c2');
+
+           // Initial state of the new control
+           expect(currentStateOf([c2])).toEqual([
+             {errors: null, pending: true, status: 'PENDING'},  // Control 2
+           ]);
+
+           expect(logs.length).toBe(0);
+
+           g1.setControl('one', c2);
+
+           tick(1);
+
+           expect(logs).toEqual([
+             'value (g1): {"one":"new!"}',  // value change emitted by `setControl`
+             'status (g1): PENDING',        // value change emitted by `setControl`
+             'status (c2): INVALID',        // async validator run after `setControl` call
+             'status (g1): PENDING',        // async validator run after `setControl` call
+             'status (g1): INVALID'         // async validator run after `setControl` call
```

Comment:
I'd propose updating comments a bit:
```suggestion
             'value (g1): {"one":"new!"}',
             'status (g1): PENDING',    // g1 async validator is invoked after `g1.setControl` call
             'status (c2): INVALID',      // c2 async validator trigger at c2 init, completed with the `INVALID` status
             'status (g1): PENDING',    // c2 update triggered g1 to re-run validation
             'status (g1): INVALID'       // g1 validator completed with the `INVALID` status
```